src/leaderboard/messages/momento_request.rs

```diff
@@ -0,0 +1,17 @@
+use crate::{Leaderboard, MomentoResult};
+
+/// A trait that allows Momento request types to define their interaction with the gRPC client.
+pub trait MomentoRequest {
+    #[allow(missing_docs)]
+    type Response;
+
+    /// An internal fn that allows Momento request types to define their interaction with
+    /// the gRPC client. You can impl this fn for your own types if you'd like to hand them
+    /// to the Momento client directly, but that is not an explicitly supported scenario and
+    /// this signature may change a little over time. If that's okay with you, impl away!
```

Comment:
This guidance is untrue, for reasonable interpretations. `next_data_client` is crate-private, and `send` is expected to implement actually sending.

Additionally, this is not how momento should support users plugging in to (currently called) MomentoRequest. Users should `impl From<MyOwnStruct> for LeaderboardFetchRequest` and use `leaderboard.send(my_own.into())` or something to that effect.

Reply:
I've clarified in [2e155bd](https://github.com/momentohq/client-sdk-rust/pull/438/commits/2e155bd21c33ff9a887f04865841f73664ce968b). I copied this comment from the cache client. Looking at that, I'm not sure why we would to make it pluggable. You're right that unless we make `next_data_client` public, then the end user has no way of sending the request in their implementation.